app/src/main/java/app/igormatos/botaprarodar/presentation/main/users/UsersFragment.kt

```diff
@@ -47,41 +53,41 @@ class UsersFragment : androidx.fragment.app.Fragment(), UsersAdapter.UsersAdapte
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        itemAdapter = UsersAdapter(this)
         binding.btnRegisterUsers.setOnClickListener {
             val intent = Intent(it.context, UserFormActivity::class.java)
             startForResult.launch(intent)
         }
-        setupRecyclerView()
-        getUsers(preferencesModule.getJoinedCommunity().id)
         binding.tieSearchUsers.addTextChangedListener {
-            itemAdapter.filter.filter(it.toString())
+            usersAdapter.filter.filter(it.toString())
         }
-    }
-
-    private fun getUsers(joinedCommunityId: String) {
-        firebaseHelperModule.getUsers(joinedCommunityId, false, object : RequestListener<Item> {
-            override fun onChildChanged(result: Item) {
-                itemAdapter.updateItem(result as User)
-            }
-
-            override fun onChildAdded(result: Item) {
-                itemAdapter.addItem(result as User)
-            }
-
-            override fun onChildRemoved(result: Item) {
-                itemAdapter.removeItem(result as User)
-            }
-        })
+        setupRecyclerView()
+        getUsers()
+        observers()
     }
 
     private fun setupRecyclerView() {
         val marginTop = resources.getDimensionPixelSize(R.dimen.padding_medium)
         rv_users.layoutManager = LinearLayoutManager(context)
-        rv_users.adapter = itemAdapter
+        rv_users.adapter = usersAdapter
         rv_users.addItemDecoration(UserDecoration(marginTop))
     }
 
+    private fun getUsers() {
+        val joinedCommunityId = preferencesModule.getJoinedCommunity().id
+        usersViewModel.getUsers(joinedCommunityId)
+    }
+
+    private fun observers() {
```

Comment:
que tal colocar um nome mais descritivo com a ação, como initObservers ou startObservers?

```suggestion
    private fun initObservers() {
```